pkg/options/options.go

```diff
@@ -84,6 +88,13 @@ func (o *Options) AddFlags() {
 	o.flags.Var(&o.Namespaces, "namespace", fmt.Sprintf("Comma-separated list of namespaces to be enabled. Defaults to %q", &DefaultNamespaces))
 	o.flags.Var(&o.MetricWhitelist, "metric-whitelist", "Comma-separated list of metrics to be exposed. This list comprises of exact metric names and/or regex patterns. The whitelist and blacklist are mutually exclusive.")
 	o.flags.Var(&o.MetricBlacklist, "metric-blacklist", "Comma-separated list of metrics not to be enabled. This list comprises of exact metric names and/or regex patterns. The whitelist and blacklist are mutually exclusive.")
+	o.flags.Int32Var(&o.Shard, "shard", int32(0), "The instances shard nominal (zero indexed) within the total number of shards. (default 0)")
+	o.flags.IntVar(&o.TotalShards, "total-shards", 1, "The total number of shards. Sharding is disabled when total shards is set to 1.")
+
+	autoshardingNotice := "When set, it is expected that --pod and --pod-namespace are both set. This is used for auto-detecting sharding. If set, this has preference over statically configured sharding. This is experimental, it may be removed without notice."
+
+	o.flags.StringVar(&o.Pod, "pod", "", "Name of the pod. "+autoshardingNotice)
+	o.flags.StringVar(&o.Namespace, "pod-namespace", "", "Name of the namespace. "+autoshardingNotice)
```

Comment:
Will this maybe `""` default to `default` namespace, because k8s? 🤔 :)

Reply:
It doesn’t, that’s kubectl Logic doing that. Here we’re referencing a concrete object so this isn’t correct.